provider/query_parameter.go

```diff
@@ -0,0 +1,58 @@
+package provider
+
+import (
+	"fmt"
+	"strings"
+)
+
+// QueryParameter represents an HTTP query parameter specified for use with
+// a given map instance.
+type QueryParameter struct {
+	Name  string `toml:"name"`
+	Token string `toml:"token"`
+	Type  string `toml:"type"`
+	SQL   string `toml:"sql"`
+	// DefaultSQL replaces SQL if param wasn't passed. Either default_sql or
+	// default_value can be specified
+	DefaultSQL   string `toml:"default_sql"`
+	DefaultValue string `toml:"default_value"`
```

Comment:
What do you think about supporting `required`? Or should we assume that query parameters are necessarily optional? I feel like supporting `required` is simple enough and might be useful in some cases.